src/repositories/user.repository.ts

```diff
@@ -30,4 +41,41 @@ export class UserRepository extends DefaultCrudRepository<
       async () => orderRepository,
     );
   }
+
+  async login(credentials: Credentials) {
+    // Validate Email
+    if (!isemail.validate(credentials.email)) {
+      throw new HttpErrors.UnprocessableEntity('invalid email');
+    }
+
+    // Validate Password Length
+    if (credentials.password.length < 8) {
+      throw new HttpErrors.UnprocessableEntity(
+        'password must be minimum 8 characters',
+      );
+    }
+
+    // Check if user exists
+    try {
+      const foundUser = await this.findOne({
+        where: {email: credentials.email, password: credentials.password},
+      });
+
+      let token = undefined;
+      if (foundUser) {
+        const currentUser = _.pick(toJSON(foundUser), ['id', 'email']);
+        // Generate user token using JWT
+        token = await signAsync(currentUser, 'secretforjwt', {
+          expiresIn: 300,
+        });
+      } else {
+        return Promise.reject(
+          new HttpErrors.Unauthorized('Wrong credentials!'),
+        );
+      }
+      return Promise.resolve(token);
+    } catch (err) {
+      return Promise.reject(new Error('Wrong Credentials!'));
```

Comment:
This code is throwing away all details of the original error (`err`), including the error message and stack trace, and reporting a generic error with no details. This is an anti-pattern, such code makes it very difficult to troubleshoot errors. Especially if they happen only in production on every other Friday night.

Async functions forward any unhandled errors as rejected promises, no extra code is needed here.